modules/configuration/R5.0/webui.nix

```diff
@@ -0,0 +1,382 @@
+config.storageManager.authProtocol = "http";
+config.storageManager.apiVersion = ["v1"];
+config.storageManager.strictSSL = false;
+config.storageManager.ca = "";
+
+// VNConfig API server and port.
+config.cnfg = {};
+config.cnfg.server_ip = ["127.0.0.1"];
+config.cnfg.server_port = "8082";
+config.cnfg.authProtocol = "http";
+config.cnfg.strictSSL = false;
+config.cnfg.ca = "";
+config.cnfg.statusURL = "/global-system-configs";
+
+// Analytics API server and port.
+config.analytics = {};
+config.analytics.server_ip = ["127.0.0.1"];
+config.analytics.server_port = "8081";
+config.analytics.authProtocol = "http";
+config.analytics.strictSSL = false;
+config.analytics.ca = "";
+config.analytics.statusURL = "/analytics/uves/bgp-peers";
+
+//DNS API Server and port.
+/* Please note: being introspect port, SSL options for dns should come from
+   config.introspect.ssl configuration
+ */
+config.dns = {};
+config.dns.server_ip = ["127.0.0.1"];
+config.dns.server_port = "8092";
+config.dns.statusURL = "/Snh_PageReq?x=AllEntries%20VdnsServersReq";
+
+// vcenter related parameters
+config.vcenter = {};
+config.vcenter.server_ip = "127.0.0.1";         //vCenter IP
+config.vcenter.server_port = "443";             //Port
+config.vcenter.authProtocol = "https";          //http or https
+config.vcenter.datacenter = "vcenter";          //datacenter name
+config.vcenter.dvsswitch = "vswitch";           //dvsswitch name
+config.vcenter.strictSSL = false;               //Validate the certificate or ignore
+config.vcenter.ca = "";                         //specify the certificate key file
+config.vcenter.wsdl = "${contrailPkgs.webui.webui-core}/webroot/js/vim.wsdl";
+
+/*****************************************************************************
+ * The below configurations descibe the SSL options for connecting to different
+ * introspect port.
+ *
+ * enabled:
+ *      Boolean flag to enable or disable ssl while connecting to different
+ *      introspect port
+ * key:
+ *      Private key to use for SSL
+ * cert:
+ *      Public x509 certificate to use
+ * ca:
+ *      A string, Buffer or array of strings or Buffers of trusted certificates
+ *      in PEM format. These are used to authorize connections.
+ * strictSSL:
+ *      If true, the server certificate is verified against the list of
+ *      supplied CAs
+ *****************************************************************************/
+config.introspect = {};
+config.introspect.ssl = {};
+config.introspect.ssl.enabled = false;
+config.introspect.ssl.key = "";
+config.introspect.ssl.cert = "";
+config.introspect.ssl.ca = "";
+config.introspect.ssl.strictSSL = false;
+
+/* Job Server */
+config.jobServer = {};
+config.jobServer.server_ip = "127.0.0.1";
+config.jobServer.server_port = "3000";
+
+/* Upload/Download Directory */
+config.files = {};
+config.files.download_path = "/tmp";
+
+/* Cassandra Server */
+config.cassandra = {};
+config.cassandra.server_ips = ["10.0.1.4"];
```

Comment:
We should use localhost here I think. This IP doesn't exist in the test VM.

```
Apr 29 08:18:59 machine 7d2msi7kmyi412r6kkmhnr01n16qs6h6-unit-script-contrail-web-server-start[2102]: { [Error: All host(s) tried for query failed. First host tried, 10.0.1.4:9042: Error: Connection timeout. See innerErrors.]
Apr 29 08:18:59 machine 7d2msi7kmyi412r6kkmhnr01n16qs6h6-unit-script-contrail-web-server-start[2102]:   innerErrors:
Apr 29 08:18:59 machine 7d2msi7kmyi412r6kkmhnr01n16qs6h6-unit-script-contrail-web-server-start[2102]:    { '10.0.1.4:9042':
Apr 29 08:18:59 machine 7d2msi7kmyi412r6kkmhnr01n16qs6h6-unit-script-contrail-web-server-start[2102]:       { [Error: Connection timeout]
Apr 29 08:18:59 machine 7d2msi7kmyi412r6kkmhnr01n16qs6h6-unit-script-contrail-web-server-start[2102]:         message: 'Connection timeout',
Apr 29 08:18:59 machine 7d2msi7kmyi412r6kkmhnr01n16qs6h6-unit-script-contrail-web-server-start[2102]:         info: 'Cassandra Driver Error' } },
Apr 29 08:18:59 machine 7d2msi7kmyi412r6kkmhnr01n16qs6h6-unit-script-contrail-web-server-start[2102]:   info: 'Represents an error when a query cannot be performed because no host is available or could be reached by the driver.',
Apr 29 08:18:59 machine 7d2msi7kmyi412r6kkmhnr01n16qs6h6-unit-script-contrail-web-server-start[2102]:   message: 'All host(s) tried for query failed. First host tried, 10.0.1.4:9042: Error: Connection timeout. See innerErrors.' }
```

Reply:
Oh, thanks.Good catch